### PR TITLE
Bulk Domain Transfer: Improve domain form

### PR DESCRIPTION
**Pull request by escapemanuele**
[Designs](mbuq4o5QxohQpBU9YoexXo-fi-405%3A9731)

## Changes

- Column for price
- Fixed authorization code error message
- Changed copy
- Popover
- Total price

## Testing
1. Live link
2. Reach `setup/domain-transfer/domains`
3. Test the step

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx [hunk 1]
```diff
@@ -39,6 +47,36 @@ const domainInputFieldIcon = ( isValidDomain: boolean, shouldReportError: boolea
...
 };
 
+const DomainPrice = ( { rawPrice, saleCost, currencyCode }: DomainPriceProps ) => {
+	const { __ } = useI18n();
+
+	if ( ! rawPrice ) {
+		return <div className="domains__domain-price-number disabled">$0</div>;
+	}
+
+	if ( ! saleCost && saleCost !== 0 ) {
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx:67]:
```suggestion
	if ( saleCost === 0 ) {
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx [hunk 2]
```diff
@@ -39,6 +47,36 @@ const domainInputFieldIcon = ( isValidDomain: boolean, shouldReportError: boolea
 	);
 };
 
+const DomainPrice = ( { rawPrice, saleCost, currencyCode }: DomainPriceProps ) => {
+	const { __ } = useI18n();
+
+	if ( ! rawPrice ) {
+		return <div className="domains__domain-price-number disabled">$0</div>;
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx]: We should probably call `formatCurrency` on the zero.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx (hunk 1 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx:67]: I thought about this, but what about sales that are not zero but half the price? 🤔

Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx:67]: You're right. Bad suggestion.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx [hunk 3]
```diff
@@ -88,8 +92,26 @@ const Domains: React.FC< Props > = ( { onSubmit } ) => {
 		}
 	};
 
+	const totalPrice = Object.values( domainsState ).reduce( ( total, currentDomain ) => {
+		if ( currentDomain.saleCost ) {
+			return total + currentDomain.saleCost;
+		}
+
+		return total + currentDomain.rawPrice;
+	}, 0 );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx]: This probably should be a function outside the component, takes the state and gives a formatted price with a correct currency. I.e shouldn't be locked to USD.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx [hunk 4]
```diff
@@ -39,6 +57,40 @@ const domainInputFieldIcon = ( isValidDomain: boolean, shouldReportError: boolea
 	);
 };
 
+const DomainPrice = ( { rawPrice, saleCost, currencyCode = 'USD' }: DomainPriceProps ) => {
+	const { __ } = useI18n();
+
+	if ( ! rawPrice ) {
+		return (
+			<div className="domains__domain-price-number disabled">
+				{ formatCurrency( 0, currencyCode, { stripZeros: true } ) }
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx]: I just realized that we don't need currency for the zero, `0` is enough, but not `$0`.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx [hunk 5]
```diff
@@ -107,6 +129,9 @@ const Domains: React.FC< Props > = ( { onSubmit } ) => {
 			domain: '',
 			auth: '',
 			valid: false,
+			rawPrice: 0,
+			saleCost: undefined,
+			currencyCode: 'USD',
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx]: Should be `undefined`.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx [hunk 6]
```diff
@@ -143,6 +168,10 @@ const Domains: React.FC< Props > = ( { onSubmit } ) => {
 					{ __( 'Add another domain' ) }
 				</Button>
 			) }
+			<div className="bulk-domain-transfer__total-price">
+				<div>{ __( 'Total' ) }</div>
+				<div>{ formatCurrency( totalPrice, 'USD', { stripZeros: true } ) }</div>
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domains.tsx]: Should be covered by the function I mention above.

**Review by alshakero — CHANGES_REQUESTED** [review]
Amazing work. This has been much harder than it seems. Left some comments.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx (hunk 4 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx]: Ok! Do you mean just having
`<div className="domains__domain-price-number disabled">0</div>`
right?

**Comment by escapemanuele** [comment]
Thank you @alshakero, I think I addressed all your feedback!

File: client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx (hunk 4 above)
Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx]: Yes!

**Review by alshakero — APPROVED** [review]
Pushed some changes and it looks great now!

**Review by alshakero — APPROVED** [review]
Pushed some changes and it looks great now!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8549986

Hi @escapemanuele, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Unique code proving ownership, needed for secure domain transfer between registrars.`
- `Add your domains`

Thank you in advance!

**Comment by cometgrrl** [comment]
Minor copy feedback:
“Add as little or as many”
I think “few” goes with “many” (silly, but English is :weird: like that)
Text wrapping is not great. Maybe putting the second sentence on a new line?
SCR-20230713-kvbk

**Comment by cometgrrl** [comment]
From the checkout screen, the back button used to go to /setup/domain-transfer/domains, and now it goes to /start/domains, which seems wrong.

**Comment by john-legg** [comment]
This is great work! While I was testing, I noticed that the width of the Price column is a bit wide. The link to the designs doesn't seem to work for me, so I don't know if this is intentional or not, but I just wanted to point it out. It's more noticeable when there's multiple domain rows.

Screenshot 2023-07-13 at 5 59 34 PM

**Comment by escapemanuele** [comment]
Hey @john-legg, thanks for the feedback. I gave the price column some space to accomodate for this:

image

The `First year free` is taking some space.
This is from the designs:

image

I'll try to remove some space 😄

**Comment by karenroldan** [comment]
Hi, @escapemanuele! 

**issue (ux,non-blocking):** There is a dollar sign on the price per domain based on Figma. But, right now, it only shows `0`

**Comment by escapemanuele** [comment]
Thank you @karenroldan, we'll try to address this!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8549986).

**Pull request merged**